analysis.py

```diff
@@ -0,0 +1,44 @@
+import json
+class Analysis:
+    def __init__(self, config):
+        self.log = self.loadJSON(config["logfile"])
+        self.numTribes = config["environmentMaxTribes"]
+        self.tribeSummary = {}
+
+    def findDominantTribe(self):
+        tribeKeys = self.getTribeKeys()
+
+        assimilate = False
+        for day in self.log:
+            for tribe in tribeKeys:
+                if self.isTribeDominant(day[tribe], day["population"]) == True:
+                    self.tribeSummary["dominantTribe"] = tribe
+                    self.tribeSummary["timestep"] = day["timestep"]
+                    self.tribeSummary["totalPopulation"] = day["population"]
+                    assimilate = True
+                    break
+            if assimilate == True:
+                break
+
+    def getTribeKeys(self):
+        tribeKeys = []
+        keysList = list(self.log[0].keys())
+        for key in keysList:
+            if "Tribe" in key:
+                tribeKeys.append(key)
+        return tribeKeys
+
+    def isTribeDominant(self, tribe, totalPopulation):
```

Comment:
The assimilation flag activates within the first few timesteps because we're checking if any tribe has more than an equal proportion. Even if there is 1 more agent in a tribe than expected (ex. 49:51 agents), the tribe will be considered dominant. I know we're still trying to figure out what metric to use for assimilation—I just wanted to give an update on the current metric.